### DISCUSSION TITLE
OBAL and STM32 i2c Slave Interface

### DISCUSSION THREAD

**Discussion by jkieffer72** [HefnySco/OBAL#38] [General]
Greetings.  I recently learned of OBAL and then saw the videos with an STM32 "black pill" being used as an i2c slave with connections for RC Input/Output ADC.  I am very interested in this code for integration into some of my projects.

I have been riding the struggle bus so far with some detours along the way.  I did manage to get ArduRover built and installed on my Pi4b, along with the sensors and assorted modules.  

I have also built several target firmwares for some existing flight controllers I was hoping to recycle.  I compared the code/settings for a couple of F7 and F4 based boards I have.  (OMNIBUSF4, OMNIBUSF7, JHEF7DUAL)   I think I am on the right track, but have not been able to make a connection between the Pi and the STM32.    THe firmware(s) for the controllers seems to compile and flash OK.  However, I am not seeing the device when I run i2cdetect.  I have had a few variations of not working.  EIther No i2c devices are detected, or my compass/baro are detected but no STM32 (x66) listed.   or, I get a device listed at every possible address.  

I get similar results across the three boards I have tried.  I am sure I have something incorrectly programmed, just need to figure out what.  

Has anyone experienced similar and have a fix?    I am not sure how exactly to share and submit my changes for review.  I have to focus elsehwere for a while, but will be back to this later today and into the weekend and will update if things change.

**Comment by HefnySco** [comment]
I hope you find my article useful:

[https://medium.com/@mohammad.hefny/ardupilot-linux-based-board-stm32-as-a-breakout-module-5b57989863ce](https://medium.com/@mohammad.hefny/ardupilot-linux-based-board-stm32-as-a-breakout-module-5b57989863ce)
